model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -306,18 +351,18 @@ message ProcessBundleProgressRequest {
   string instruction_id = 1;
 }
 
-// A request to provide full MonitoringInfo for a given id.
+// A request to provide full MonitoringInfo for a given bundle.
 //
 // An SDK can report metrics using an identifier that only contains the
 // associated payload. A runner who wants to receive the full metrics
 // information can request all the monitoring metadata via a
 // MonitoringInfosMetadataRequest providing a list of ids as necessary.
 //
-// The MonitoringInfo ids are scoped to the associated control connection. For
-// example, an SDK may reuse the ids across multiple bundles.
+// The SDK is allowed to reuse the identifiers across multiple bundles as long
+// as the MonitoringInfo could be reconstructed fully by overwriting its
+// payload field with the bytes specified here.
 message MonitoringInfosMetadataRequest {
-  // A list of ids for which MonitoringInfo are requested. All but the payload
-  // field will be populated.
+  // A list of ids for which the full MonitoringInfo is requested for.
   repeated string monitoring_info_id = 1;
 }
 
```

Comment:
Can you copy the update to say `for the lifetime of the associated control connection` in ProcessBundleProgressResponse and ProcessBundleResponse

Reply:
I believe I have applied all of your suggestions now. I did some through this webpage and some locally.